runtime/doc/lsp.txt

```diff
@@ -1469,6 +1462,68 @@ save({lenses}, {bufnr}, {client_id})                 *vim.lsp.codelens.save()*
       • {client_id}  (integer)
 
 
+==============================================================================
+Lua module: vim.lsp.inlay_hint                                *lsp-inlay_hint*
+
+get({opts})                                         *vim.lsp.inlay_hint.get()*
+    Get the list of inlay hints, (optionally) restricted by buffer, client, or
+    range. This method does not make an LSP request, but just queries locally
+    stored inlay hints. The returned inlay hints are unresolved.
+
+    Example usage: >lua
+        local timeout, bufnr = 100, vim.api.nvim_get_current_buf()
+        local hint = vim.lsp.inlay_hint.get({ bufnr = bufnr })[1]
+
+        local client = vim.lsp.get_clients({ id = hint.client_id })[1]
+        resolved_hint = client.request_sync(
+          'inlayHint/resolve', hint.inlay_hint, timeout, bufnr).result
+        vim.lsp.util.apply_text_edits(resolved_hint.textEdits, bufnr, 'utf-8')
+
+        local label_part = resolved_hint.label[1]
+        command = label_part.command
+        vim.lsp.buf.execute_command({ command = command })
```

Comment:
This is kinda wrong/ incomplete as command can refer to client commands.
we've a client. _exec_cmd function that does the right thing but it's private so far. 

Maybe we can remove this part for now? We definitely shouldn't spread this incorrect pattern further.  In fact I think we should deprecate buf.execute_command because it's inherently broken. A command always originates from a single server and buf.execute_command broadcasts it to all clients on the attached buffers

Reply:
Thanks for the tip!